### PR TITLE
added suport an context package  and added service update and getall posts #6

### PR DESCRIPTION
**Pull request by aclgo**

### CODE REVIEW

**Comment by filhodanuvem** [comment]
We had to merge a new PR which caused conflicts with this one @aclgo . I think this PR will be more about the new endpoints now, thanks for adding those with test cases.

**Comment by aclgo** [comment]
conflict resolved

File: internal/http/actions.go [hunk 1]
```diff
@@ -65,12 +79,20 @@ func DeletePosts(ctx *gin.Context) {
 
 func GetPosts(ctx *gin.Context) {
 	param := ctx.Param("id")
-	id, err := uuid.Parse(param)
```
Comment (filhodanuvem) [inline internal/http/actions.go:68]: why did you remove the uuid type support? Don't you think that we can avoid reaching out the database if the user is sending a non uuid ? (eg.: `GET /posts/filhodanuvem`

Comment (aclgo) [reply internal/http/actions.go:68]: realmente, não pensei nisso

**Comment by filhodanuvem** [comment]
well done @aclgo , thank you for the help
